File: src/org/thoughtcrime/securesms/util/DelimiterUtil.java
package org.thoughtcrime.securesms.util;


import android.text.TextUtils;

import java.util.regex.Pattern;

public class DelimiterUtil {

  public static String escape(String value, char delimiter) {
    return value.replace("" + delimiter, "\\" + delimiter);
  }

  public static String unescape(String value, char delimiter) {
    return value.replace("\\" + delimiter, "" + delimiter);
  }

  public static String[] split(String value, char delimiter) {
    if (TextUtils.isEmpty(value)) {
      return new String[0];
    } else {
      String regex = "(?<!\\\\)" + Pattern.quote(delimiter + "");
      return value.split(regex);
    }
  }
}

File: libsession/src/main/java/org/session/libsession/utilities/DelimiterUtil.kt
package org.session.libsession.utilities

import java.util.regex.Pattern

object DelimiterUtil {
    @JvmStatic
    fun escape(value: String, delimiter: Char): String {
        return value.replace("" + delimiter, "\\" + delimiter)
    }

    @JvmStatic
    fun unescape(value: String, delimiter: Char): String {
        return value.replace("\\" + delimiter, "" + delimiter)
    }

    @JvmStatic
    fun split(value: String, delimiter: Char): List<String> {
        val regex = Regex("(?<!\\\\)" + Pattern.quote(delimiter + ""))
        return value.split(regex)
    }
}